common-ts/src/common-ui-utils/TransactionErrorHandler.ts

```diff
@@ -0,0 +1,357 @@
+import { Ref } from '../utils';
+import {
+	JupV4Errors,
+	JupV6Errors,
+	DriftCompetitionsErrors,
+	DriftErrors,
+} from '../constants/errorCodes';
+import { PrettyError } from '../types';
+
+type AnchorError = { code?: number; logs?: string[] } & Error & {
+		error: { code: number; message: string };
+	};
+
+type NonDriftProgramErrorHandlerConfig = {
+	errorsList: {
+		[name: string]: {
+			code: number;
+			name: string;
+			msg: string;
+		};
+	};
+	errorToastTitle: string;
+};
+
+// todo - build these up as we go
+const NON_DRIFT_CODES_INDICATING_TX_VERSION_ISSUE = ['-32603'];
+const DRIFT_CODES_INDICATING_TX_VERSION_ISSUE = [];
+
+const NON_DRIFT_PROGRAMS_ERROR_HANDLER_CONFIGS: NonDriftProgramErrorHandlerConfig[] =
+	[
+		{
+			errorsList: JupV4Errors.errorsList,
+			errorToastTitle: 'Jupiter Swap Error',
+		},
+		{
+			errorsList: JupV6Errors.errorsList,
+			errorToastTitle: 'Jupiter Swap Error',
+		},
+		{
+			errorsList: DriftCompetitionsErrors.errorsList,
+			errorToastTitle: 'Drift Competitions Error',
+		},
+	];
+
+type TransactionHandlerOpts = {
+	showErrorLogs?: boolean;
+	notifyCallback?: (props: any) => void;
+	captureExceptionCallback?: (exception: Error) => void;
+	isDev?: boolean;
+};
+
+export class TransactionErrorHandler {
+	showErrorLogs: boolean;
+	notifyCallback?: (prop: any) => void;
```

Comment:
this is gonna be a super hairy one, for when we want to integrate this into our other UIs 🥲

created a linear card for this: https://linear.app/driftprotocol/issue/DPE-1668/properly-abstract-notification-callback-in-common-tss

Reply:
did this btw, just used current notify props as the props for now